web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx

```diff
@@ -0,0 +1,421 @@
+      />
+      {!hasIamPermError && tableData.currRegion && (
+        <>
+          <Box mb={2}>
+            <Toggle
+              isToggled={isAppDiscoveryEnabled}
+              onToggle={() => setAppDiscoveryEnabled(!isAppDiscoveryEnabled)}
+            >
+              <Box ml={2} mr={1}>
+                Enable Kubernetes App Discovery
+              </Box>
+              <ToolTipInfo>
+                Teleport's Kubernetes App Discovery will automatically identify
+                and enroll to Teleport HTTP applications running inside a
+                Kubernetes cluster.
+              </ToolTipInfo>
+            </Toggle>
+          </Box>
+          <ClustersList
+            items={tableData.items}
+            fetchStatus={tableData.fetchStatus}
+            selectedCluster={selectedCluster}
+            onSelectCluster={setSelectedCluster}
+            fetchNextPage={fetchNextPage}
+          />
+          <StyledBox mb={5} mt={5}>
+            <Text mb={2}>Automatically enroll selected EKS cluster</Text>
+            <ButtonSecondary
+              width="215px"
+              type="submit"
+              onClick={handleOnEnroll}
+              disabled={enrollmentNotAllowed}
+              mt={2}
+              mb={2}
+            >
+              Enroll EKS Cluster
+            </ButtonSecondary>
+            <Box>
+              <ButtonText
+                disabled={enrollmentNotAllowed}
+                onClick={() => {
+                  setIsManualHelmDialogShown(b => !b);
+                }}
+                pl={0}
+              >
+                Or click here to see instructions for manual enrollment
+              </ButtonText>
+            </Box>
+          </StyledBox>
+        </>
+      )}
+      {hasIamPermError && (
+        <Box mb={5}>
+          <ConfigureIamPerms
+            kind="eks"
+            region={tableData.currRegion}
+            integrationRoleArn={agentMeta.awsIntegration.spec.roleArn}
+          />
+        </Box>
+      )}
+      {(enrollmentState.status === 'enrolling' ||
+        enrollmentState.status === 'error') && (
+        <EnrollmentDialog
+          clusterName={selectedCluster.name}
+          close={closeEnrollmentDialog}
+          error={enrollmentState.error}
+          status={enrollmentState.status}
+        />
+      )}
+      {isManualHelmDialogShown && (
+        <ManualHelmDialog
+          command={command}
+          cancel={() => setIsManualHelmDialogShown(false)}
+          confirmedCommands={() => {
+            setEnrollmentState({ status: 'awaitingAgent' });
+            setIsManualHelmDialogShown(false);
+            setIsAgentWaitingDialogShown(true);
+          }}
+        />
+      )}
+      {isAgentWaitingDialogShown && (
+        <AgentWaitingDialog
+          joinToken={joinToken}
+          status={enrollmentState.status}
+          clusterName={selectedCluster.name}
+          setWaitingResult={(result: Kube) => {
```

Comment:
normally `setXXX` naming pattern is used for react state `const [reactState, setReactState] = React.useState`, so i would rename this, maybe `updateWaitingResult`?